### PR TITLE
remove old share propagation entries from appconfig

### PR DESCRIPTION
**Pull request by icewind1991**
Cleanup step for #22720

cc @MorrisJobke @rullzer @PVince81

### CODE REVIEW

File: lib/private/repair/sharepropagation.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	public function getName() {
+		return 'Remove old share propagation app entries';
+	}
+
+	public function run() {
+		$keys = $this->config->getAppKeys('files_sharing');
+
+		foreach ($keys as $key) {
```
Comment (MorrisJobke) [inline lib/private/repair/sharepropagation.php]: Really? From a performance point of view this is really bad :(

Comment (MorrisJobke) [inline lib/private/repair/sharepropagation.php]: I would rather fire a `DELETE FROM oc_appconfig WHERE configkey = "files_sharing";`

Comment (rullzer) [inline lib/private/repair/sharepropagation.php]: NO! That would remove a lot more settings as well.

Comment (icewind1991) [inline lib/private/repair/sharepropagation.php]: can't do, there are entries in there which we shouldn't delete

Comment (MorrisJobke) [inline lib/private/repair/sharepropagation.php]: Ah - I missed the numeric check. :(

Comment (MorrisJobke) [inline lib/private/repair/sharepropagation.php]: Sadly then we need to go that way

Comment (nickvergessen) [inline lib/private/repair/sharepropagation.php]: Well it should be "one last time" where we have to do this horrible query.

**Comment by nickvergessen** [comment]
:+1:

**Comment by PVince81** [comment]
See also https://github.com/owncloud/core/issues/20944 to stop the abuse

**Comment by PVince81** [comment]
So this means that after running an upgrade, all people who had pending propagations will be missing them ?

**Comment by PVince81** [comment]
Ok, now I remember. These are only timestamps from the last check. So after an upgrade the "are there any shares to propagate ?" check will be redone when the values are missing.

Let's test this.

**Comment by PVince81** [comment]
Hmmm, I can't make these values appear again. @icewind1991 are they even useful any more in 9.0 since the new propagation ?

**Comment by icewind1991** [comment]
> are they even useful any more in 9.0 since the new propagation ?

No, they are no longer created/user, hence the cleanup

**Comment by PVince81** [comment]
Okay, then this makes sense. Thanks for the clarification :+1:

**Comment by PVince81** [comment]
@icewind1991 can you rebase ? Would be good to have OCI tests pass since this PR touches DB stuff.
Thanks.

**Comment by icewind1991** [comment]
rebased

**Comment by rullzer** [comment]
:+1:

**Comment by PVince81** [comment]
Steps to create the DB entries in OC 8.2

1. Create three users "user1", "user2", "user3"
1. Login as "user1"
1. Create a folder "test"
1. Share "test" with "user1"
1. Login as "user2"
1. Share "test" with "user3"
1. Login as "user3"
1. Create a file "test/test.txt"
1. `select * from oc_appconfig where appid='files_sharing'`, see the values with numeric key
1. Edit/save the file "test/test.txt"
1. `select * from oc_appconfig where appid='files_sharing'`, see the values with numeric key, the values have changed to more recent timestamps
1. Upgrade to OC 9.0.5
1. Edit/save the file "test/test.txt"
1. `select * from oc_appconfig where appid='files_sharing'`, see the values with numeric key, the values did **not** change
1. Delete the values from the DB (or alternatively re-run the above steps directly on OC 9.0.5)
1. Edit/save the file "test/test.txt"
1. => no more numeric values in oc_appconfig

@DeepDiver1975 @butonic

**Comment by PVince81** [comment]
these entries are not needed any more because since OC 9.0 the etag is computed based on the root etag of all child storages here: https://github.com/owncloud/core/blob/v9.0.5/lib/private/files/fileinfo.php#L167
`childEtag` is gathered here: https://github.com/owncloud/core/blob/v9.0.5/lib/private/files/fileinfo.php#L336

This logic didn't exist in OC <= 8.2: https://github.com/owncloud/core/blob/v8.2.8/lib/private/files/fileinfo.php#L148

So I think we should backport this repair step to OC 9.0.x.

@DeepDiver1975 @butonic

**Comment by PVince81** [comment]
stable9: https://github.com/owncloud/core/pull/26454

**Pull request merged**